layer updates to make use of newer capabilities

adam-stokes opened the pull request:
- add apt layer and let our packages be managed by that
- other cleanups

Signed-off-by: Adam Stokes battlemidget@users.noreply.github.com

Commit:
layer updates to make use of newer capabilities

- add apt layer and let our packages be managed by that
- other cleanups

Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>

battlemidget commented:
@chuckbutler @marcoceppi please provide feedback and any changes you require

lazypower commented on README.md:
```diff
@@ -7,18 +7,37 @@ Include `layer:nginx` in your `layer.yaml`
 The variables in `site.toml` will be available when configuring the NGINX
 `templates/vhost.conf`.
 
+If additional packages are required (ie php-mysql, php-mcrypt) in your `layer.yaml` add:
```
I am +1 to this change... i wonder if this shouldn't be a config option however, so the end user can change the dependencies at deploy time rather than enforcing a build time construct

lazypower replied:
Actually I would probably invert this. NGINX is the one known package we expect to come with this layer, so move that to the layer.yaml install bits, and make the extra packages part of charm config, and use the apt layer to install the extra packages.

chuckbutler commented:
I had some minor nits here about usability in where we are installing the packages, but otherwise I think this is a good change. If you agree with my comments, I'm happy to +1 once those have been inverted.

Commit:
add nginx-full in layer apt config

Extra packages can still be handled by additional apt definitions in
consumers layer.yaml

Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>

lazypower reviewed (approved):
+1 LGTM
